javascript/ql/src/Security/CWE-693/InsecureHelmet.ql

```diff
@@ -0,0 +1,58 @@
+/**
+ * @name Insecure configuration of Helmet security middleware
+ * @description The Helmet middleware is used to set security-related HTTP headers in Express applications. This query finds instances where the middleware is configured with important security features disabled.
+ * @kind problem
+ * @problem.severity error
+ * @security-severity 5.0
+ * @precision high
+ * @id javascript/insecure-helmet-configuration
+ * @tags security
+ *        cwe-693
+ *        cwe-1021
+ */
+
+import semmle.javascript.frameworks.ExpressModules
+
+class HelmetProperty extends Property {
+  ExpressLibraries::HelmetRouteHandler helmet;
+
+  HelmetProperty() {
+    helmet.(DataFlow::CallNode).getAnArgument().asExpr().(ObjectExpr).getAProperty() = this
+  }
+
+  ExpressLibraries::HelmetRouteHandler getHelmet() { result = helmet }
+
+  predicate isFalse() { this.getInit().(BooleanLiteral).getBoolValue() = false }
+
+  predicate isImportantSecuritySetting() {
+    this.getName() in ["frameguard", "contentSecurityPolicy"]
+    or
+    // read from data extensions to allow enforcing other settings
+    requiredHelmetSecuritySetting(this.getName())
+  }
+}
```

Comment:
Here is a suggestion for how you can use some more modern libraries.  (Staying in the dataflow layer).  

```suggestion
class HelmetProperty extends DataFlow::Node instanceof DataFlow::PropWrite {
  ExpressLibraries::HelmetRouteHandler helmet;

  HelmetProperty() {
    this = helmet.(DataFlow::CallNode).getAnArgument().getALocalSource().getAPropertyWrite()
  }

  ExpressLibraries::HelmetRouteHandler getHelmet() { result = helmet }

  predicate isFalse() { DataFlow::PropWrite.super.getRhs().mayHaveBooleanValue(true) }

  string getName() { result = DataFlow::PropWrite.super.getPropertyName() }

  predicate isImportantSecuritySetting() {
    this.getName() in ["frameguard", "contentSecurityPolicy"]
    or
    // read from data extensions to allow enforcing other settings
    requiredHelmetSecuritySetting(this.getName())
  }
}
```